pkg/registry/authorization/selfsubjectaccessreview/rest.go

```diff
@@ -73,6 +76,11 @@ func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation
 	}
 
 	var authorizationAttributes authorizer.AttributesRecord
+	if len(userToCheck.GetExtra()[authorizationv1.ScopesKey]) > 0 {
+		// the self-SAR requires user:full scope in order to perform the access review
+		userToCheck.GetExtra()[authorizationv1.ScopesKey] = []string{authorizationscope.UserFull}
```

Comment:
this will likely mutate the user in the context, we shouldn't do that

Reply:
we now create a new user.Info object when we need to update the scopes